appserver/website-bootstrap/explore.js

```diff
@@ -173,73 +177,159 @@ $(document).ready(function(){
+    if(!fromDate && !toDate){ // if no dates are provided, don't need to add from or to parameters to the url
+        fillTemporalTable(url);
+    } else if(fromDate && !toDate){ // if a start date is provided but not an end date
+        startButNoEndDate(fromDate, url);
+    } else if(!fromDate && toDate){ // if an end date is provided but not a start date
+        alert("If you enter a To Date, please also enter a From Date!");
+    } else { // if both a start date and an end date is provided
+        startAndEndDate(fromDate, toDate, url);
+    }
+}
+
+function startAndEndDate(start, end, url) {
+    let startDate = new Date(start);
+    let startBoundary = new Date("02/29/2020");
+    let endDate = new Date(end);
+    let endBoundary = Date.now();
+
+    if(startDate < startBoundary || endDate > endBoundary) {
+        alert("Please enter valid  dates between March 1, 2020 and today!");
+    } else if(startDate > endDate){
+        alert("Please make sure that the end date is after the start date!");
+    } else {
+        let startDateStr = formatStart(startDate);
+        url += '&from=' + startDateStr; //YYYY-mm-dd HH:MM:SS
+        let endDateStr = formatEnd(endDate);
+        url += '&to=' + endDateStr; //YYYY-mm-dd HH:MM:SS
+        fillTemporalTable(url);
+    }
+}
+
+function startButNoEndDate(start, url) {
+    let startDate = new Date(start);
+    let startBoundary = new Date("02/29/2020");
+
+    if(startDate < startBoundary) {
+        alert("Please enter a valid start date between March 1, 2020 and today!");
+    } else {
+        let startDateStr = formatStart(startDate);
+        url += '&from=' + startDateStr; //YYYY-mm-dd HH:MM:SS
+        fillTemporalTable(url);
+    }
+}
+
+// function endButNoStartDate(end, url) {
+//     let endDate = new Date(end);
+//     let endBoundary = Date.now();
+//
+//     if(endDate > endBoundary) {
+//         alert("Please enter a valid end date between March 1, 2020 and today!");
+//     } else {
+//         let endDateStr = formatEnd(endDate);
+//         url += '&to=' + endDateStr; //YYYY-mm-dd HH:MM:SS
+//         fillTemporalTable(url);
+//     }
+// }
+
+function formatStart(start){
+    let startDateStr = start.toISOString();
+    startDateStr = startDateStr.replace("T", " ");
+    //cuts off the milliseconds and the 'Z' at the end of the string
+    let millisPeriodIndex = startDateStr.lastIndexOf(".");
+    startDateStr = startDateStr.substring(0, millisPeriodIndex);
+
+    startDateStr = encodeURIComponent(startDateStr);
+    return startDateStr;
+}
+
+function formatEnd(end){
+    let endDateStr = end.toISOString();
+        endDateStr = endDateStr.replace("T", " ");
+        //cuts off the milliseconds and the 'Z' at the end of the string
+        let millisPeriodIndex = endDateStr.lastIndexOf(".");
+        endDateStr = endDateStr.substring(0, millisPeriodIndex);
+
+        endDateStr = encodeURIComponent(endDateStr);
+        return endDateStr;
```

Comment:
why are these functions separate, instead of having one function 'formatDateForAPI()' (or something like that)?

Reply:
true! will fix!

Reply:
good to go!